Lib/test/test_descr.py

```diff
@@ -1967,7 +1967,7 @@ def test_methods_in_c(self):
         # different error messages.
         set_add = set.add
 
-        expected_errmsg = "descriptor 'add' of 'set' object needs an argument"
+        expected_errmsg = "set.add() needs an argument"
```

Comment:
Agree, this is confusing.

Reply:
I can change the error message to anything you like, except that it must contain the string `set.add` (the function name). So it could be
```
TypeError: unbound method set.add() needs an argument
```
or whatever (surely, this is better than anything mentioning descriptors).

------------

Small rant: the bug is really this:
```
>>> set().add()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: add() takes exactly one argument (0 given)
```
For a Python method, it would correctly note that 2 arguments are required. This is difficult to fix in CPython since `builtin_function_or_method` doesn't really know whether it's a function or method. This was one of the things that the rejected PEP 580 would have addressed.